src/ProgramRepresentation/DisjointedPVAliasSets.cpp

```diff
@@ -0,0 +1,125 @@
+#include "ProgramRepresentation/DisjointedPVAliasSets.h"
+#include "Debug.h"
+
+typename std::list<PVAliasSet>::iterator
+DisjointedPVAliasSets::findIter(ProgramVariable programVar) {
+    for (auto it = sets.begin(); it != sets.end(); ++it) {
+        if (it->contains(programVar)) {
+            return it;
+        }
+    }
+
+    return sets.end();
+}
+
+void DisjointedPVAliasSets::merge(
+    typename std::list<PVAliasSet>::iterator set1,
+    typename std::list<PVAliasSet>::iterator set2) {
+    if (set1 != sets.end() && set2 != sets.end()) {
+        set1->programVariables.splice(set1->programVariables.end(),
+                                      set2->programVariables);
+        sets.erase(set2);
+    }
+}
+bool DisjointedPVAliasSets::elementIsInAnySet(ProgramVariable programVar) {
+    return findIter(programVar) != sets.end();
+}
+
+std::list<PVAliasSet> DisjointedPVAliasSets::getSets() const {
+    return sets;
+}
+
+PVAliasSet *DisjointedPVAliasSets::getSetRef(ProgramVariable programVar) {
+    return getSetRef(programVar.getCleanedName());
+}
+
+PVAliasSet *DisjointedPVAliasSets::getSetRef(const std::string& cleanedName) {
+    for (PVAliasSet &set : sets) {
+        if (set.contains(cleanedName)) {
+            return &set;
+        }
+    }
+
+    return NULL;
+}
+
+PVAliasSet *DisjointedPVAliasSets::getSetRef(Value* val) {
+    for (PVAliasSet &set : sets) {
+        if (set.contains(val)) {
+            return &set;
+        }
+    }
+
+    return NULL;
+}
+
+void DisjointedPVAliasSets::unionSets(ProgramVariable elementA,
+                                      ProgramVariable elementB) {
+    auto set1 = findIter(elementA);
+    auto set2 = findIter(elementB);
+
+    if (set1 == set2 || set1 == sets.end() || set2 == sets.end()) {
+        return;
+    }
+
+    merge(set1, set2);
+}
+void DisjointedPVAliasSets::makeSet(ProgramVariable programVar) {
+    PVAliasSet newSet;
+    newSet.add(programVar);
+    sets.push_back(newSet);
```

Comment:
I think this method should have some validation logic so that it fails if `programVar` is already in a set, since if that precondition doesn't hold calling this method will break the data structure. Even if it makes the implementation a bit slower, it will be worth it if we ever fail to check that precondition before we call this method (which will probably happen eventually as the code base evolves).